src/osemgrep/cli_scan/Core_runner.ml

```diff
@@ -181,16 +181,96 @@ let runner_config_of_conf (conf : conf) : Runner_config.t =
         version = Version.version;
       }
 
+let pp_table (h1, heading) ppf entries =
+  let int_size i =
+    let rec dec acc = function
+      | 0 -> acc
+      | n -> dec (succ acc) (n / 10)
+    in
+    dec 1 i
+  in
+  let len1, lengths =
+    let acc = List.map String.length heading in
+    List.fold_left
+      (fun (n1, needed) (c1, curr) ->
+        ( max (String.length c1) n1,
+          List.map2 max needed (List.map int_size curr) ))
+      (String.length h1, acc)
+      entries
+  in
+  let llen = List.fold_left (fun acc w -> acc + w + 1) len1 lengths in
+  let line = String.make llen '-' in
+  let pad =
+    let max_len = List.fold_left max len1 lengths in
+    let spaces = String.make max_len ' ' in
+    fun str_size len ->
+      let to_pad = succ (len - str_size) in
+      String.sub spaces 0 to_pad
+  in
+  Fmt.pf ppf "%s%s" h1 (pad (String.length h1) len1);
+  List.iter2
+    (fun h l -> Fmt.pf ppf "%s%s" h (pad (String.length h) l))
+    heading lengths;
+  Fmt.pf ppf "@.%s@." line;
+  List.iter
+    (fun (e1, entries) ->
+      Fmt.pf ppf "%s%s"
+        (String.capitalize_ascii e1)
+        (pad (String.length e1) len1);
+      List.iter2
+        (fun e l -> Fmt.pf ppf "%s%u" (pad (int_size e) l) e)
+        entries lengths;
+      Fmt.pf ppf "@.")
+    entries;
+  Fmt.pf ppf "@."
+
+let pp_heading ppf txt =
```

Comment:
We probably want to move those helper functions to a separate file. maybe src/osemgrep/reporting/Fmt_helpers.ml

Reply:
With also a Fmt_helpers.mli with some basic documentation about each helper function (ex of output for example).

Reply:
done in [8532ca4](https://github.com/returntocorp/semgrep/pull/7445/commits/8532ca47b40bcf67365ecfc2be078e8db31311d9)